ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -6031,6 +6046,23 @@ IHqlExpression * WorkflowTransformer::extractWorkflow(IHqlExpression * untransfo
         if ((info.persistOp != no_persist) && expr->isAction())
             setValue.setown(transformSequentialEtc(setValue));
 
+         if(info.persistOp == no_critical)
+         {
+            StringBuffer criticalName;
+            info.storedName->queryValue()->getStringValue(criticalName);
+            unsigned criticalWfid = ++wfidCount;
+            Owned<IWorkflowItem> wf = addWorkflowToWorkunit(wfid, WFTypeNormal, WFModeCritical, queryDirectDependencies(setValue), conts, info.queryCluster());
+            setWorkflowCritical(wf, criticalName.str());
+
+            DependenciesUsed dependencies(false);
```

Comment:
This code in the persist processing is there because there is an extra workflow item to calculate whether the persist needs rebuilding.  That isn't needed for the critical sections.